mlrun/feature_store/retrieval/local_merger.py

```diff
@@ -61,6 +61,7 @@ def _generate_vector(
                 columns={name: alias for name, alias in columns if alias}, inplace=True
             )
             dfs.append(df)
+            del df
```

Comment:
Same question here.

Reply:
As discussed, it's to avoid keeping the last df after merge and before write to targets.